test/CMakeLists.txt

```diff
@@ -1,5 +1,5 @@
-cmake_minimum_required ( VERSION 3.13.0 )
-project ( "CoreMQTT unit test"
+cmake_minimum_required ( VERSION 3.22.0 )
+project ( "CoreMQTT tests"
           VERSION 1.0.0
```

Comment:
Is this version incorrect at this point?
Know it's unrelated, but just noticed it

Reply:
In my opinion, I think it should be separate from the versioning of the library

The reason I think that is that I feel it provides a good way of knowing when functionally breaking changes happen in the library? I think it makes it clear that "Library <VERSION> change from <A.B.C> to <C.D.E> caused a functionality change. This caused the tests to move from <U.V.W> to <X.Y.Z>"

Reply:
Makes sense. I do have some different point of views though: If we were to version this same as the library, then we need not keep track of which version of tests work with which version of library. It will be the same as the library.

Additionally, since these tests are in the same repository, I don't think that anyone would choose a different/older set of tests to run against this code base. If the tests were in a separate repository, then I think properly versioning the tests make much more sense.

What do you think?

Reply:
> If we were to version this same as the library ... It will be the same as the library.

Which I would say is the single best reason to keep the versioning the same, makes it easy to understand that a version of the tests are meant for a single version of the source code and that is all we test.

> ... then we need not keep track of which version of tests work with which version of library.

And I wouldn't have much desire to maintain a matrix of what versions of the tests work with which version of the source code. But I think it can be useful for regression testing to know "These versions of the code work with these versions of the tests"

> Additionally, since these tests are in the same repository, I don't think that anyone would choose a different/older set of tests to run against this code base. 

Part of why I think they should be versioned differently is because of some of our repos that don't have the tests in them with the source code. For example the [FreeRTOS-Kernel's](https://github.com/FreeRTOS/FreeRTOS-Kernel) tests live inside of the FreeRTOS/FreeRTOS repository in the [FreeRTOS/Test](https://github.com/FreeRTOS/FreeRTOS/tree/main/FreeRTOS/Test) directory. However I recognize this isn't a huge deal, just means that a separate PR needs to be created when updating the version in the kernel repo.


But all that said, I'm more than happy to do either if somebody feels strongly about it.